dali/python/nvidia/dali/pipeline.py

```diff
@@ -990,3 +996,93 @@ def iter_setup(self):
         For example, one can use this function to feed the input
         data from NumPy arrays."""
         pass
+
+
+def _discriminate_args(func, **func_kwargs):
```

Comment:
I wonder how it plays with political correctness ;P

Reply:
I mentioned it before but the comment got lost.
- I'd use "split" or "separate", discriminate might carry a negative connotation.
- There is a function ```def _separate_kwargs(kwargs):``` in ops.py. Isn't this exactly what you need? (haven't read it)

Reply:
There's a discriminator in electronics, mathematics, GANs also have a discriminator. I wouldn't like to be oversensitive - should we avoid to use "gap", just because there is a wage-gap?

(sacrasm warning)
After all, that's precisely what this function does - treats function args better than ctor args ;)

Reply:
Achievement unlocked: troll ;)

Reply:
How about the existing function I pointed out? Is it a duplicate or not?